### DISCUSSION TITLE
Bring back from background

### DISCUSSION THREAD

**Discussion by ghost** [input-leap/input-leap#1754] [Feature Requests]
> This issue has been migrated from old Barrier Github repository https://github.com/debauchee/barrier/issues/495
> 
> Issue created on: 2019-11-16 by @zd-dalibor
> Issue last updated on: 2021-09-02

Is there are any way to bring back the barrier application from a background in gnome-shell?

There is an option to show a tray icon when the application is minimized but as you already know gnome-shell has abandoned this feature. 

Also with this [KStatusNotifierItem/AppIndicator Support](https://extensions.gnome.org/extension/615/appindicator-support/) gnome extension, I have no luck even if most of the other applications tray icons are shown with this extension (like skype...).

The only way to start a new barrier instance properly is to kill previous one (there are two processes in the background `barrier` and `barrierc`).

Also, barrier clients cannot go to sleep (start screen server or turn off the monitor).

---

> Commented on: 2020-10-01 by @github-actions[bot]

Is this issue still an issue for you? Please do comment and let us know! Alternatively, you may close the issue yourself if it is no longer an problem

**Comment by liselorev** [comment]
I can confirm this issue is still applicable.

My configuration is

 barrier CLIENT on "Windows 10"
 barrier CLIENT on "MacOS Ventura"

 SERVER on "Ubuntu 22.04.3 LTS"
barrier installed via snap package (refers to version 2.4.0+dfsg / application indicates "2.4.0-release-ac5a1bfd" )

When initial startup (on the server machine), the gui / dialog is presented.
When closing that dialog window, it closes, but the server application/version remains running (and connection remains functional).
 09:23 ?        00:00:02 barrier
 09:23 ?        00:00:53 /snap/barrier/682/usr/bin/barriers -f --no-tray --debug INFO --name queenie --disable-crypto --disable-client-cert-checking -c /tmp/Barrier.TJ5154 --address :25442

however when relaunching the "app", a second instance is created, and the dialog/gui fails to connect (since the original server is still running (and maintaining the connection)).

So on the server,... I fail to have access (after closing the app) to the loggin of the connection... since the new instance has its own (failing to start) logging.

**Comment by shymega** [comment]
Hi,

Thanks for reporting your experience. It is worth noting that you seem to have experienced this on the *`Barrier`* software package, not Input Leap. This makes the experience invalid, as there have been many changes to the codebase since the fork.

I would kindly request that you test on Input Leap, and report back.

Many thanks.
